### ISSUE TITLE
Try using backup locks instead of FTWRL on MariaDB

### ISSUE THREAD

**Issue by willbryant** [willbryant/kitchen_sync#77]
https://jira.mariadb.org/browse/MDEV-5336 (merged in MariaDB 10.4.2) introduced a new backup lock system with a number of backup stages; block_commit could be enough to get a consistent read position for multiple workers without needing to use `FLUSH TABLES WITH READ LOCK` like we do now.

FTWRL blocks until running queries and open transactions finish, and itself blocks new statements from starting - so it's very disruptive to run on an active server.  Avoiding it would be highly desirable in terms of safety.

MySQL has introduced LOCK INSTANCE FOR BACKUP (https://dev.mysql.com/doc/refman/8.0/en/lock-instance-for-backup.html) based more closely on the Percona Server backup lock model, but at a glance that doesn't have the same commit-blocking behavior.

(PostgreSQL doesn't have this issue because it uses non-blocking read snapshot export & import rather than locking, which is better again.)

**Comment by willbryant** [comment]
Confirmed that MySQL's `LOCK INSTANCE FOR BACKUP` would not do the job, so this would be a mariadb-specific improvement.

**Comment by willbryant** [comment]
I tried this out in origin/mariadb_backup_locks 384d34c.

It turns out to be a nontrivial thing to implement support for, as the first connection seems to lose its read transaction snapshot when BACKUP STAGE END is run.  So we'd need to use a dummy connection (or a dummy worker, though that would have compatibility concerns).

That's definitely doable.  However, while testing it out, I discovered that as a part of the work to implement backup locks, MariaDB 10.4 also significantly improved the behavior of FLUSH [NO_WRITE_TO_BINLOG] TABLES [WITH READ LOCK] - as of https://github.com/MariaDB/server/commit/163b34fe25919b25ff83860f30f2440b44c8b53b, it no longer blocks against long-running SELECT statements!  Nor does it block against open write _transactions_; just currently running write _statements_.

My experience is that in practice, long-running DML isn't all that super common.  So at this time, it seems like the main benefits are achieved just by upgrading to MariaDB 10.4, and the additional benefit from KS implementing backup locks would be pretty small.

I'm not ruling it using backup locks for the future, but since it's nontrivial to implement and the benefit is quite limited on those versions it would actually work on, I'm going to go ahead and close this ticket.  Will reopen if someone shows that significant problems remain.
